rules/opentitan.bzl

```diff
@@ -43,15 +45,166 @@ PER_DEVICE_DEPS = {
+        CONST.LCV.TEST_UNLOCKED4,
+        CONST.LCV.TEST_UNLOCKED5,
+        CONST.LCV.TEST_UNLOCKED6,
+        CONST.LCV.TEST_UNLOCKED7,
+        CONST.LCV.RMA,
+    ])
+
+def create_dev_key(name, label):
+    return create_key_(name, label, [
+        CONST.LCV.DEV,
+    ])
+
+def create_prod_key(name, label):
+    return create_key_(name, label, [
+        CONST.LCV.TEST_UNLOCKED0,
+        CONST.LCV.TEST_UNLOCKED1,
+        CONST.LCV.TEST_UNLOCKED2,
+        CONST.LCV.TEST_UNLOCKED3,
+        CONST.LCV.TEST_UNLOCKED4,
+        CONST.LCV.TEST_UNLOCKED5,
+        CONST.LCV.TEST_UNLOCKED6,
+        CONST.LCV.TEST_UNLOCKED7,
+        CONST.LCV.DEV,
+        CONST.LCV.PROD,
+        CONST.LCV.PROD_END,
+        CONST.LCV.RMA,
+    ])
+
+def create_keyset(rsa_key, spx_key):
+    return struct(
+        rsa = rsa_key,
+        spx = spx_key,
+    )
+
+# Keys available in the repo
+SILICON_CREATOR_KEYS = struct(
+    RSA = struct(
+        FAKE = struct(
+            TEST = [
+                create_test_key("rsa_fake_test_key_0", "@//sw/device/silicon_creator/rom/keys/fake:test_private_key_0"),
+            ],
+            DEV = [
+                create_dev_key("rsa_fake_dev_key_0", "@//sw/device/silicon_creator/rom/keys/fake:dev_private_key_0"),
+            ],
+            PROD = [
+                create_prod_key("rsa_fake_prod_key_0", "@//sw/device/silicon_creator/rom/keys/fake:prod_private_key_0"),
+            ],
+        ),
+        REAL = None,
+        UNAUTHORIZED = [
+            create_key_("rsa_unauthorized_0", "@//sw/device/silicon_creator/rom/keys:unauthorized_private_key_0", []),
+        ],
+    ),
+    SPX = struct(
+        FAKE = struct(
+            TEST = [
+                create_test_key("spx_fake_test_key_0", "@//sw/device/silicon_creator/rom/keys/fake/spx:test_key_0_spx"),
+            ],
+            DEV = [
+                create_dev_key("spx_fake_dev_key_0", "@//sw/device/silicon_creator/rom/keys/fake/spx:dev_key_0_spx"),
+            ],
+            PROD = [
+                create_prod_key("spx_fake_prod_key_0", "@//sw/device/silicon_creator/rom/keys/fake/spx:prod_key_0_spx"),
+            ],
+        ),
+        REAL = None,
+        UNAUTHORIZED = [
+            create_key_("spx_unauthorized_0", "@//sw/device/silicon_creator/rom/keys/spx:unauthorized_0_spx", []),
+        ],
+    ),
+)
+
+SILICON_OWNER_KEYS = struct(
+    RSA = struct(
+        FAKE = struct(
+            TEST = [
+                create_test_key("rsa_fake_rom_ext_test_key_0", "@//sw/device/silicon_creator/rom_ext/keys/fake:rom_ext_test_private_key_0"),
+            ],
+            DEV = [
+                create_dev_key("rsa_fake_rom_ext_dev_key_0", "@//sw/device/silicon_creator/rom_ext/keys/fake:rom_ext_dev_private_key_0"),
+            ],
+            PROD = None,
+        ),
+        REAL = None,
+        UNAUTHORIZED = None,
+    ),
+)
+
+def zip(l1, l2):
+    length = min(len(l1), len(l2))
+    return [(l1[i], l2[i]) for i in range(length)]
```

Comment:
I think `zip` is a starlark builtin: https://docs.bazel.build/versions/5.4.0/skylark/lib/globals.html#zip